core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -37,6 +37,22 @@ function formatDuration(milliseconds) {
   return hours.toFixed(1) + " h";
 }
 
+function makeIdNumeric(id) {
+  var strs = id.split("-");
```

Comment:
The application ids are actually "_".  However if running in local mode then it is "-". local-1455658099053.  We should check is any other formats.